sklearn/metrics/pairwise.py

```diff
@@ -1566,6 +1566,13 @@ def additive_chi2_kernel(X, Y=None):
     return result
 
 
+@validate_params(
+    {
+        "X": ["array-like"],
+        "Y": ["array-like", None],
+        "gamma": [Interval(Real, None, None, closed="neither"), Hidden(np.ndarray)],
```

Comment:
```suggestion
        "gamma": [Real, Hidden(np.ndarray)],
```
I think.

Reply:
``Interval(Real, None, None, closed="neither")`` excludes ``-inf``, ``+inf`` and ``nan``
``Interval(Real, None, None, closed="both")`` inclues ``-inf`` and ``+inf`` but excludes ``nan``
``Real`` includes ``-inf``, ``+inf`` and ``nan``

I don't think we want to make inf valid for gamma, do we ?

Reply:
haha, that's quite subtle, not very clear by looking at the code. Thanks @jeremiedbb